### PR TITLE
#3803 ADR Modal

### LINKED ISSUE #3803: DispensingPage - new ADR modal
## Is your feature request related to a problem? Please describe.

Following from https://github.com/openmsupply/mobile/issues/3802 - create a new modal using `JSONForm` which will create an ADR

## Describe the solution you'd like

As anove

## Implementation

**A description of how the solution would best be implemented in the code base**

## Describe alternatives you've considered

**A clear and concise description of any alternative solutions or features you've considered.**

## Additional context

**Add any other context or screenshots about the feature request here.**

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #3803 

## Change summary

- Adds an ADR modal to create new ADRs

## Testing

**NOTE: Need some footrunner for an ADR form- will becoming soon**

- [ ] Can create a new ADR, save it and sync it to the server - checking in the record browser, all the fields are filled correctly
- [ ] Cannot create an ADR which is not valid

### CODE REVIEW

**Comment by joshxg** [comment]
used 

```
$formSchema:=ds.form_schema.all().first()



$adrSchema:=ds.form_schema.new()
$adrSchema.ID:=UUID_generate
$adrSchema.type:="ADR"
$adrSchema.json_schema:=$formSchema.json_schema
$adrSchema.ui_schema:=$formSchema.ui_schema
$adrSchema.version:=1
$adrSchema.save()
```

to just use my survey form

**Comment by joshxg** [comment]
image

File: src/widgets/modalChildren/ADRInput.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import { selectCurrentPatient } from '../../selectors/patient';
+
+export const ADRInputComponent = ({ onCancel, onSave, patient }) => {
+  const [{ formData, isValid }, setForm] = useState({ formData: null, isValid: false });
+
+  return (
+    <>
+      <JSONForm
+        onChange={(changed, validator) => {
+          setForm({ form: changed.formData, isValid: validator(changed.formData) });
```
Comment (mark-prins) [inline src/widgets/modalChildren/ADRInput.js:22]:
```suggestion
          setForm({ formData: changed.formData, isValid: validator(changed.formData) });
```

**Review by mark-prins — CHANGES_REQUESTED** [review]
A typo in the `ADRInput` - with that changed, it's all working for me!

**Comment by joshxg** [comment]
🙄 sorry again 🤣

**Review by mark-prins — APPROVED** [review]
🎉

**Comment by mark-prins** [comment]
> 🙄 sorry again 🤣

I appreciated it.. makes me feel like I'm being useful

**Pull request merged**